torch/csrc/jit/script/compiler.cpp

```diff
@@ -1314,6 +1317,34 @@ struct to_ir {
     emitLoopCommon(stmt.range(), {}, {cond}, stmt.body(), {});
   }
 
+
+  // Currently we do not support assigning exceptions to variables,
+  // a = Exception("hi")
+  // raise a
+  //
+  // We don't validate that the name of the Exception
+  // raise AnyName(3)
+  //
+  // NYI: add exception logic to control-flow nodes
+  // if True:
+  //   a = 1
+  // else
+  //   raise Exception("Hi")
+  // print(a)
+  void emitRaise(const Raise& stmt) {
+    if (stmt.expr().kind() != TK_APPLY) {
+      throw ErrorReport(stmt) << "Invalid exception. Exceptions cannot "
+        "be constructed outside of a raise statement";
+    }
+    auto apply = Apply(stmt.expr());
+    auto exception_name = Var(apply.callee()).name().name();
```

Comment:
Now you are allowing anything that must be a call node in the parser. But you are not forcing the thing being called to be a variable. This now has the same problem as above. Just ignore the entire exception statement.